toolkit/ThreeWToolkit/data_visualization/correlation_heatmap.py

```diff
@@ -0,0 +1,61 @@
+from typing import cast
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+
+from .base_visualizer import BaseVisualizer
+
+
+class CorrelationHeatmap(BaseVisualizer):
+    def __init__(
+        self,
+        df_of_series: pd.DataFrame,
+        **kwargs,
+    ) -> None:
+        self.df_of_series = df_of_series
+        self.kwargs = kwargs
+
+    def plot(self, ax: Axes | None = None) -> tuple[Figure, Axes]:
+        title: str = self.kwargs.pop("title", "Correlation Heatmap")
+        figsize: tuple = self.kwargs.pop("figsize", (10, 8))
+
+        if ax is None:
+            fig, ax = plt.subplots(figsize=figsize)
+        else:
+            fig = cast(Figure, ax.get_figure())
+
```

Comment:
For the other plots you checked if the df contains only NaN to prevent an empty plot.
Maybe it would be good to do the same here:

```suggestion
        clean_series = self.df_of_series.dropna()
        if clean_series.empty:
            raise ValueError("Series contains only NaN values")
```